File: other/StarStairs.java
/*
Given a number n, print stairs of stars of n steps. Use recursion.

E.g.
Input: 4
Output:
*
**
***
****

*/
public class StarStairs{	
	public static void printStairs(int num){
		if(num==0) return; //base condition
		printStairs(num-1);
		for(int i=0;i<num;i++){
			System.out.print("*");
		}
		System.out.println();
	}
}

File: other/StarStairs.cs
/*
Given a number n, print stairs of stars of n steps. Use recursion.

E.g.
Input: 4
Output:
*
**
***
****

*/
public static partial class Solution{
	public static void StarStairs(int num){
		PrintStairs(num);
		return;
	}
	private static void PrintStairs(int num){
		if(num==0){ //base condition			
			return; 
		}		
		PrintStairs(num-1);
		for(int i=0;i<num;i++){
			Console.Write("*");
		}
		Console.WriteLine();
	}
}